cmd/profilecli/blocks_tool.go

```diff
@@ -0,0 +1,365 @@
+package main
+
+import (
+	"context"
+	"embed"
+	_ "embed"
+	"fmt"
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/grafana/dskit/server"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"golang.org/x/exp/slices"
+
+	phlareobj "github.com/grafana/pyroscope/pkg/objstore"
+	objstoreclient "github.com/grafana/pyroscope/pkg/objstore/client"
+	"github.com/grafana/pyroscope/pkg/objstore/providers/gcs"
+	"github.com/grafana/pyroscope/pkg/phlaredb/block"
+	"github.com/grafana/pyroscope/pkg/phlaredb/bucket"
+	"github.com/grafana/pyroscope/pkg/phlaredb/bucketindex"
+	httputil "github.com/grafana/pyroscope/pkg/util/http"
+)
+
+//go:embed tool/blocks/tool.blocks.index.gohtml
+var indexPageHtml string
+
+//go:embed tool/blocks/tool.blocks.list.gohtml
+var blocksPageHtml string
+
+//go:embed tool/blocks/tool.blocks.detail.gohtml
+var blockDetailsPageHtml string
+
+//go:embed tool/static
+var staticFiles embed.FS
+
+type blocksWebToolParams struct {
+	httpListenPort  int
+	objectStoreType string
+	bucketName      string
+}
+
+func addBlocksWebToolParams(blocksWebToolCmd commander) *blocksWebToolParams {
+	var (
+		params = &blocksWebToolParams{}
+	)
+	blocksWebToolCmd.Flag("object-store-type", "The type of the object storage (e.g., gcs).").Default("gcs").StringVar(&params.objectStoreType)
+	blocksWebToolCmd.Flag("bucket-name", "The name of the object storage bucket.").StringVar(&params.bucketName)
```

Comment:
I really would prefer if we could mimic flags from pyroscope itself and support all the options by calling `RegisterFlags`.

Unsure that might be also incredibly hard

Reply:
I tried but it doesn't seem simple enough to do without a bigger refactor, can plan this for later though.